docs/06_how-to-guides/05_authorization/how_to_restrict_access_to_an_action_by_user.md

```diff
@@ -1,13 +1,41 @@
 ---
-content_title: How to restrict access to an action by a user
+content_title: How To Perform Authorization Checks
+link_text: How To Perform Authorization Checks
 ---
 
 ## Preconditions
-- It is assumed you have the sources for a contract and one of the actions defined is getting as a parameter an account name and it is printing the account name.
 
-To restrict access to the `hi` action, you can do it in two ways:
+It is assumed the following:
+
+1. You have the sources of a contract with one of the actions defined, let's call it `hi` action.
+2. The `hi` action has defined one input parameter `user` of type `name`.
+3. The `hi` action is printing the name of the `user` account.
+4. The `hi` action needs to authorize the `user` account.
+
+## Authorization Methods
+
+To restrict access to the `hi` action, you can do it in three ways.
+
+### 1. Using eosio::check(eosio::has_auth(...)...)
+
+The below code is enforcing the action `hi` to be executed only by the account that is sent as parameter to the action, no matter what permission the account is using to sign the transaction (e.g. owner, active, code).
```

Comment:
Use simple present tense in technical docs (the account uses). Avoid present continuous tense (the account is using):
"The below code is enforcing" --> "The code below enforces"
"The account is using" --> "The account uses"
In general, gerunds (-ing verb terminations) should be avoid if possible.

Reply:
thanks.